iothub/device/src/Convention/CommandRequest.cs

```diff
@@ -42,17 +43,32 @@ internal CommandRequest(PayloadConvention payloadConvention, string commandName,
         public string CommandName { get; }
 
         /// <summary>
-        /// The command request payload.
+        /// The command request payload, deserialized to the specified type.
         /// </summary>
         /// <typeparam name="T">The type to deserialize the command request payload to.</typeparam>
-        /// <returns>The command request payload.</returns>
-        public T GetPayload<T>()
+        /// <param name="payload">When this method returns true, this contains the value of the command request payload.
+        /// When this method returns false, this contains the default value of the type <c>T</c> passed in.</param>
+        /// <returns><c>true</c> if the command request payload can be deserialized to type <c>T</c>; otherwise, <c>false</c>.</returns>
+        public bool TryGetPayload<T>(out T payload)
         {
-            string dataAsJson = GetPayloadAsString();
+            payload = default;
+            string jsonPayload = GetPayloadAsString();
 
-            return dataAsJson == null
-                ? default
-                : _payloadConvention.PayloadSerializer.DeserializeToType<T>(dataAsJson);
+            try
+            {
+                if (jsonPayload == null)
+                {
+                    return false;
+                }
+
+                payload = _payloadConvention.PayloadSerializer.DeserializeToType<T>(jsonPayload);
+                return true;
+            }
+            catch (Exception)
+            {
+            }
+
+            return false;
```

Comment:
To avoid an empty catch statement that looks questionable.

Reply:
It isn't the default because in this case the code has acted as a gauntlet for catching all the bad cases (above returning false where appropriate), so a `return true` at the end is the point where it has made it.

But if you prefer to just add a comment to the catch block, that's fine with me.